extensions/hp-extension/src/custom-viewport/setViewportZoomPan.ts

```diff
@@ -5,30 +5,67 @@
  * @param value 
  * @param viewport 
  */
-const setViewportZoomPan = (id,value,viewport) => {
-  const {zoom,pan, flipHorizontal, flipVertical} = value;
-  if( zoom && zoom!==1 ) viewport.setZoom(zoom);
-  if( pan ) viewport.setPan(pan);
-  if( flipHorizontal || flipVertical ) {
+const setViewportZoomPan = (id, value, viewport) => {
+  const { zoom, pan, flipHorizontal, flipVertical } = value;
+  const { imageCanvasPoint, initialDisplayArea } = value;
+
+  const camera = viewport.getCamera();
+  console.log("Camera=", JSON.stringify(camera));
+
+  // Set the base values - assume height fits for now
+  let initialZoom = 1;
+  if (initialDisplayArea) {
+    // TODO - use the correct calculation to actually fit things properly
+    // such that both initialDisplayArea's fit
+    initialZoom = 1 / Math.min(initialDisplayArea[0], initialDisplayArea[1]);
+    viewport.setZoom(initialZoom, true);
+  }
+
+  const imageData = viewport.getImageData();
+
+  if (imageCanvasPoint && imageData) {
```

Comment:
This code works like the old version, but still has a few bugs - not sure why yet - seems like the canvas size isn't quite right, or the center point isn't actually the center of the image area.